### PR TITLE
refactor: add wagmi

### PR DESCRIPTION
**Pull request by chrstph-dvx**
### Summary

Replace `useWallet` library with `wagmi`

### Steps to test

- Create a retryable transaction (tutorial [here](https://github.com/OffchainLabs/arbitrum-tutorials/tree/master/packages/redeem-failed-retryable))
- Past the transaction hash
- You should be able to connect your wallet
- Redeem button should be displayed
- On click, page should reload: transaction is successful

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,18 +38,11 @@
...
-    "development": [
-      "last 1 chrome version",
-      "last 1 firefox version",
-      "last 1 safari version"
-    ]
-  },
+  "browserslist": [
+    ">0.2%",
+    "not dead",
+    "not op_mini all"
```
[the remark this answers is not in the document]
Comment (chrstph-dvx) [reply package.json:44]: Good point, updated. Hopefully in the future, we can have a package to share those configs

File: src/WagmiProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  Chain
+} from "wagmi";
+
+import { publicProvider } from "wagmi/providers/public";
+import { jsonRpcProvider } from "wagmi/providers/jsonRpc";
+import { PropsWithChildren } from "react";
+
+const availableChains: Chain[] = [
+  ...defaultChains, // mainnet, goerli
+  ...defaultL2Chains.filter(({ id }) => id === 421613 || id === 42161) // Arbitrum One, Arbitrum goerli
```
[the remark this answers is not in the document]
Comment (chrstph-dvx) [reply src/WagmiProvider.tsx:16]: Good catch, updated

[the remark this answers is not in the document]
Comment (chrstph-dvx) [reply src/WagmiProvider.tsx:16]: Removed, see below

File: src/WagmiProvider.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+];
+
+const { provider, webSocketProvider } = configureChains(availableChains, [
```
[the remark this answers is not in the document]
Comment (chrstph-dvx) [reply src/WagmiProvider.tsx:19]: Because I didn't know about it, thanks for the link

[the remark this answers is not in the document]
Comment (chrstph-dvx) [reply src/WagmiProvider.tsx:19]: Reverted as per slack discussion

File: src/App.tsx [hunk 4]
```diff
@@ -13,10 +12,12 @@ import {
 } from "@arbitrum/sdk";
 import { useNavigate, useParams, generatePath } from "react-router-dom";
 
+import { useNetwork, useSigner } from "wagmi";
 import { ethers } from "ethers";
 import {
   JsonRpcProvider,
```
Comment (fionnachan) [inline src/App.tsx:18]: this comment is on the existing code and not sth introduced by this branch

but we should remove `JsonRpcProvider` as its only usage is for typing a variable of type `StaticJsonRpcProvider`

**Review by fionnachan — APPROVED** [review]
LGTM!

**Review by spsjvc — APPROVED** [review]
Nice, was able to test it both by redeeming a failed retryable and by claiming a withdrawal.

Approved with two small comments. 👍

**Pull request merged**